pkg/apis/kueueoperator/v1alpha1/types.go

```diff
@@ -83,12 +55,119 @@ type KueueStatus struct {
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 // KueueList contains a list of Kueue
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
 type KueueList struct {
 	metav1.TypeMeta `json:",inline"`
 	// metadata for the list
 	// +optional
 	metav1.ListMeta `json:"metadata,omitempty"`
-	// items is a slice of kueue
+	// items is a slice of Kueue
+	// this is a cluster scoped resource and there can only be 1 Kueue
+	// +kubebuilder:validation:MaxItems=1
 	// +required
 	Items []Kueue `json:"items"`
 }
+
+// +kubebuilder:validation:Enum=BatchJob;RayJob;RayCluster;JobSet;MPIJob;PaddleJob;PytorchJob;TFJob;XGBoostJob;AppWrapper;Pod;Deployment;StatefulSet;LeaderWorkerSet
+type KueueIntegration string
+
+const (
+	KueueIntegrationBatchJob        KueueIntegration = "BatchJob"
+	KueueIntegrationRayJob          KueueIntegration = "RayJob"
+	KueueIntegrationRayCluster      KueueIntegration = "RayCluster"
+	KueueIntegrationJobSet          KueueIntegration = "JobSet"
+	KueueIntegrationMPIJob          KueueIntegration = "MPIJob"
+	KueueIntegrationPaddeJob        KueueIntegration = "PaddeJob"
+	KueueIntegrationPyTorchJob      KueueIntegration = "PyTorchJob"
+	KueueIntegrationTFJob           KueueIntegration = "TFJob"
+	KueueIntegrationXGBoostJob      KueueIntegration = "XGBoostJob"
+	KueueIntegrationAppWrapper      KueueIntegration = "AppWrapper"
+	KueueIntegrationPod             KueueIntegration = "Pod"
+	KueueIntegrationDeployment      KueueIntegration = "Deployment"
+	KueueIntegrationStatefulSet     KueueIntegration = "StatefulSet"
+	KueueIntegrationLeaderWorkerSet KueueIntegration = "LeaderWorkerSet"
```

Comment:
I think you missed the `mxjob` here.

Reply:
mxjob was removed in 0.11 actually.

Reply:
https://github.com/kubernetes-sigs/kueue/issues/1429

Reply:
Ok, I'm a little confused here. Are we going to support release-0.10 at all? Or we are shaping the APIs for release-0.11?

Reply:
I think we can't release 0.10 as it doesn't have secure metrics which is a blocker for release.

Reply:
RHOAI doesn't use MXJob anyway so there isn't really anyone being impacted by this.

MXJob was deprecated because the community was no longer working on it. So dropping support for it is the right move.